packedForest/test/fpTests/fpSingletonTest.h

```diff
@@ -0,0 +1,54 @@
+#include "../../src/fpSingleton/fpSingleton.h"
+
+using namespace fp;
+
+TEST(fpSingleton, checkDataDependantParameters_errorPatchHeightMax)
```

Comment:
please add test for `paramsRandMatImagePatch`

given width = height = 28, assert top = 0
given width = height = 27, assert top = 0,1,28,29
given min/max of different sizes, sampled height/widths are within range